docs/user/agent/authentication-authorization.md

```diff
@@ -0,0 +1,124 @@
+
+1. The `auth_scheme` claim in the JWT token is extracted.
+2. If the claim is malformed or missing, an error is returned.
+3. The `auth_scheme` claim embedded in the token specifies the type of authorization (e.g., `watcherAuth`) and is compared with the `authScheme` expected by the endpoint. This ensures that the token matches the intended purpose of the request.  
+4. If the `auth_scheme` in the token does not match the `authScheme` required by the endpoint, or if an incorrect token header is provided, the request is rejected with a `Forbidden (403)` error.  
+
+For example:  
+- Suppose the endpoint `/api/assisted-install/v2/infra-envs` requires the `watcherAuth` scheme.  
+- A request is made with the following command, but an **incorrect token header** is sent:  
+  ```bash
+  curl http://localhost:8090/api/assisted-install/v2/infra-envs \
+       -H "Authorization: eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdXRoX3NjaGVtZSI6IndhdGNoZXJBdXRoIn0.hJ5EuBOttDQ1-ANWUJlc97H1k63d-clDCaFUZ84XSYdF03A938OfMU6RHWltv0AiWZEHNJ0gLrjLxR-1yUYqiA"
+  ```  
+- Although the token specifies the correct `auth_scheme` (`watcherAuth`), the **header sent is incorrect** for this request.
+
+Expected response:  
+```json
+{
+  "code": 403,
+  "message": "authClaim watcherAuth is unauthorized to access"
+}
+```
+
+**Outcome:** 
+The system identifies that the provided header does not satisfy the endpoint’s requirements and correctly responds with an error. This confirms that tokens must not only align with the expected `authScheme` but also be passed in the correct format and header. The authorization logic ensures that only authorized requests can access protected endpoints in Assisted Service.
+
+## Authentication API (Server Side)
+
+The authentication server validates the token provided by the client using the `validateToken` function. The following steps explain the server-side validation process:
+
+1. When an agent or nodes ISO is created, a set of public and private keys is generated. The private key is used to sign the JWT tokens, while the associated public key is made available to the API server.
+2. The token is passed to the API server as part of the request. The API server validates the token by verifying its signature against the public key. This ensures the token was signed by the correct private key and has not been tampered with.
+3. If the token is invalid or expired, an error is logged and the request is rejected with a `Unauthorized` error.
+4. The claims in the token are extracted.
+5. If the `exp` (expiration) claim is found, the server checks whether the token has expired.
+   - In the install workflow, the `exp` claim will not be present.
+   - In the add-nodes workflow, the token expiration is checked. If expired, a new token must be generated.
+6. If the token is valid and not expired, the claims are returned and the request proceeds.
+
+## Expired Token Handling on Day 2
```

Comment:
This entire paragraph looks like it could be integrated in the previous token management one, where the behavior for day 2 have been described